includes/blocks/class-sensei-block-take-course.php

```diff
@@ -163,34 +162,25 @@ private function render_disabled_with_prerequisite( $course_id, $content ) {
 	 *
 	 * @param int $course_id
 	 *
+	 * @deprecated 3.8.0 use Sensei_Course::get_course_prerequisite_message.
+	 *
 	 * @return string
 	 */
 	public function get_course_prerequisite_message( $course_id ) {
-		$course_prerequisite_id   = absint( get_post_meta( $course_id, '_course_prerequisite', true ) );
-		$course_title             = get_the_title( $course_prerequisite_id );
-		$prerequisite_course_link = '<a href="' . esc_url( get_permalink( $course_prerequisite_id ) )
-			. '" title="'
-			. sprintf(
-			// translators: Placeholder $1$s is the course title.
-				esc_attr__( 'You must first complete: %1$s', 'sensei-lms' ),
-				$course_title
-			)
-			. '">' . $course_title . '</a>';
-
-		$complete_prerequisite_message = sprintf(
-		// translators: Placeholder $1$s is the course title.
-			esc_html__( 'You must first complete %1$s before taking this course.', 'sensei-lms' ),
-			$prerequisite_course_link
-		);
+		_deprecated_function( __METHOD__, '3.8.0', 'Sensei_Course::get_course_prerequisite_message' );
 
-		/**
-		 * Filter sensei_course_complete_prerequisite_message.
-		 *
-		 * @param string $complete_prerequisite_message the message to filter
-		 *
-		 * @since 1.9.10
-		 */
-		return apply_filters( 'sensei_course_complete_prerequisite_message', $complete_prerequisite_message );
+		return Sensei()->course::get_course_prerequisite_message( $course_id );
+	}
 
+	/**
+	 * Helper method to avoid adding notices in the editor.
+	 *
+	 * @param string $content The notice content.
+	 * @param string $type    The notice type.
+	 */
+	private function add_notice( string $content, string $type ) {
+		if ( ! is_admin() ) {
```

Comment:
So generally `Sensei_Notices` is initialized in frontend only as notices are supposed to be used in the frontend only. 

However when using the REST API, `render_callback` is still called in order to generate the post content which is returned. This means that there might be a notice generated for the post content in the editor. I added this check to avoid initializing the code in the editor too but now that I think about it I probably should. I'll refactor.

Reply:
Done in 96d7b04d10846d0995565f327b0269c8c4b687a8